code/edgebridge/src/main/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtension.java

```diff
@@ -178,7 +210,180 @@ private void dispatchTrackRequest(final Map<String, Object> data, final Event pa
 		getApi().dispatch(event);
 	}
 
-	private boolean isNullOrEmpty(final Map map) {
-		return map == null || map.isEmpty();
+	/**
+	 * Formats track event data to the required Analytics Edge translator format under the `data.__adobe.analytics` object.
+	 *
+	 * The following is the mapping logic:
+	 * - The "action" field is mapped to "data.__adobe.analytics.linkName", and "data.__adobe.analytics.linkType" is set to "other".
+	 * - The "state" field is mapped to "data.__adobe.analytics.pageName".
+	 * - Any "contextData" keys that start with the "&&" prefix are mapped to "data.__adobe.analytics" with the prefix removed.
+	 * - Any "contextData" keys without the "&&" prefix are mapped to "data.__adobe.analytics.contextData".
+	 * - Any additional fields are passed through and left directly under the "data" object.
+	 *
+	 * As an example, the following track event data:
+	 * ```
+	 * {
+	 *    "action": "action name",
+	 *    "contextData": {
+	 *       "&&c1": "propValue1",
+	 *       "key1": "value1"
+	 *    },
+	 *    "key2": "value2"
+	 * }
+	 * ```
+	 * Is mapped to:
+	 * ```
+	 * {
+	 *   "data": {
+	 *     "__adobe": {
+	 *       "analytics": {
+	 *         "linkName": "action name",
+	 *         "linkType": "other",
+	 *         "c1": "propValue1",
+	 *         "contextData": {
+	 *           "key1": "value1"
+	 *         }
+	 *       }
+	 *     },
+	 *     "key2": "value2"
+	 *   }
+	 * }
+	 * ```
+	 *
+	 * @param data track event data
+	 * @return data formatted for the Analytics Edge translator. {@code null} if there is no data in
+	 * the payload after format rules are applied.
+	 * @throws CloneFailedException if the cloning process fails.
+	 */
+	private Map<String, Object> formatData(final Map<String, Object> data) throws CloneFailedException {
```

Comment:
with your latest changes, I think you could handle the CloseFailedException here and return null for all the error cases to the caller function.

// Create a mutable copy of data - can throw exception if deep copy fails
try {
Map<String, Object> mutableData = EventDataUtils.clone(map);
} catch () {
log + return null
}